packages/SwingSet/src/controller/initializeSwingset.js

```diff
@@ -414,16 +419,26 @@ export async function initializeSwingset(
   // kconfig.namedBundleIDs.BUNDLENAME=bundleID , which both point into
   // kconfig.idToBundle.BUNDLEID=bundle
 
+  const bundleCache = await makeNodeBundleCache(
+    'bundles/',
+    { dev: config.includeDevDependencies, format: config.bundleFormat },
+    s => import(s),
+  );
+
   async function getBundle(desc, mode, nameToBundle) {
+    trace(
+      'getBundle',
+      mode,
+      Object.keys(desc),
+      desc.moduleFormat,
+      desc.endoZipBase64Sha512 || desc.sourceSpec,
+    );
     if (mode === 'bundle') {
       return desc.bundle;
     } else if (mode === 'bundleSpec') {
       return JSON.parse(fs.readFileSync(desc.bundleSpec).toString());
     } else if (mode === 'sourceSpec') {
-      return bundleSource(desc.sourceSpec, {
-        dev: config.includeDevDependencies,
-        format: config.bundleFormat,
-      });
+      return bundleCache.load(desc.sourceSpec);
```

Comment:
@warner this is the bit we were talking about earlier today.

Reply:
The original description of #3609 by @warner has a bunch of considerations for reliable bundle caching. Before closing that one, I reduced the scope to "prototype". Using it for a dev tool has served us reasonably well, but this is a proposal to put it into production runtime, so it's worth some thought.

Reply:
It's a perf optimization so we could land this PR without it and solve the bundle caching separately.

One way would be to have the bootstrap config indicate whether to bundle cache. The test could add it to the modifications it does before passing on the config.

Another option might be to define another bundle source specifying the built bundle. Then the config would be modified to point to pre-bundled sources.

Reply:
> One way would be to have the bootstrap config indicate whether to bundle cache. The test could add it to the modifications it does before passing on the config.

I like that idea.